pkg/server/diagnostics/reporter.go

```diff
@@ -99,6 +107,11 @@ type Reporter struct {
 	// regardless of whether the response we get back is successful or
 	// not.
 	LastSuccessfulTelemetryPing atomic.Int64
+
+	client struct {
+		sync.Once
```

Comment:
Would it be better t o create a `newDiagnosticReporter` constructor that sets this field instead of using sync.Once?

Reply:
agreed. done.